agent/worker.go

```diff
@@ -39,24 +42,26 @@ type statusReporterKey struct {
 }
 
 type worker struct {
-	db        *bolt.DB
-	executor  exec.Executor
-	publisher exec.LogPublisher
-	listeners map[*statusReporterKey]struct{}
-	secrets   *secrets
+	db                *bolt.DB
+	taskevents        *events.Broadcaster
```

Comment:
This needs to be closed once it is no longer being used, otherwise it will leak a goroutine.